core/tests.py

```diff
@@ -575,3 +613,76 @@ def test_student_answered_3_times(self):
+        self.instructor = Instructor.objects.create_user(
+            username="bill.doors", password="4567"
+        )
+        self.course = Course.objects.create(name="Walls OS")
+        self.offering = Offering.objects.create(
+            course=self.course, description="Introduction to Walls OS"
+        )
+        self.enrollment = Enrollment.objects.create(
+            student=self.student, offering=self.offering
+        )
+        self.teaches = Teaches.objects.create(
+            instructor=self.instructor, offering=self.offering
+        )
+        self.exercise1 = Exercise.objects.create(
+            offering=self.offering,
+            slug="walls-exercise-1",
+            url="bill.doors.com/walls/",
+            type=Exercise.ExerciseType.CODE,
+        )
+
+        self.exercise2 = Exercise.objects.create(
+            offering=self.offering,
+            slug="walls-exercise-2",
+            url="bill.doors.com/walls/",
+            type=Exercise.ExerciseType.CODE,
+            allow_submissions=False
+        )
+    
+
+    def test_activate_already_activated_exercise(self):
+        fac = APIRequestFactory()
+        req = fac.get(f'offerings/{self.offering.pk}/exercises/{self.exercise1.slug}/activate/')
+        force_authenticate(req, self.instructor)
+
+        resp = activate_exercise(req, off_pk=self.offering.pk, ex_slug=self.exercise1.slug)
+        assert resp.status_code == 200
+        ex =  Exercise.objects.get(id=self.exercise1.pk)
+        assert ex.allow_submissions == True
+
+    def test_deactivate_already_deactivated_exercise(self):
+        fac = APIRequestFactory()
+        req = fac.get(f'offerings/{self.offering.pk}/exercises/{self.exercise2.slug}/activate/')
+        force_authenticate(req, self.instructor)
+
+        resp = deactivate_exercise(req, off_pk=self.offering.pk, ex_slug=self.exercise2.slug)
+        assert resp.status_code == 200
+        ex =  Exercise.objects.get(id=self.exercise2.pk)
+        assert ex.allow_submissions == False
+
+    def test_activate_exercise(self):
+        fac = APIRequestFactory()
+        req = fac.get(f'offerings/{self.offering.pk}/exercises/{self.exercise2.slug}/activate/')
+        force_authenticate(req, self.instructor)
+
+        resp = activate_exercise(req, off_pk=self.offering.pk, ex_slug=self.exercise2.slug)
+        assert resp.status_code == 200
+        ex =  Exercise.objects.get(id=self.exercise2.pk)
+        assert ex.allow_submissions == True
+
+    def test_deactivate_exercise(self):
+        fac = APIRequestFactory()
+        req = fac.get(f'offerings/{self.offering.pk}/exercises/{self.exercise1.slug}/deactivate/')
+        force_authenticate(req, self.instructor)
+
+        resp = deactivate_exercise(req, off_pk=self.offering.pk, ex_slug=self.exercise1.slug)
+        assert resp.status_code == 200
+        ex =  Exercise.objects.get(id=self.exercise1.pk)
+        assert ex.allow_submissions == False
```

Comment:
Maybe just add a test that checks that a student is not allowed to activate/deactivate